docs/csi-5to6-upgradepath.md

```diff
@@ -0,0 +1,27 @@
+# upgrade path from PSO 5.2.0 to PSO 6
+
+## Summary
+PSO upgrade path allows current users on PSO 5.2.0 to migrate to PSO 6. PSO 6 is stateful in that volume metadata is stored in Cockroach DB to enable PSO to provide more features and better performance for PSO users. Therefore, existing PSO CSI resources need to be migrated into Cockroach DB before they can be managed by PSO 6. 
+
+#### Usecases:
+1. upgrade from 5.2.0 to latest PSO 6 release
+
+## Restrictions
+1. Upgrade tool will **not** be migrating snapshots, snapshots content from PSO 5 to PSO 6. PSO 6 is currently leveraging beta version of K8s snapshot API and backward compatibility is not guranteed by CSI standard. That means a snapshot created in PSO 5 will not be used to restore a volume in PSO 6 after the upgrade tool. 
+2. In addition, the newly installed PSO 6 will need to be in the same Pure namespaces as PSO 5. Specifically, that means your namespace field in PSO 5 helm chart `values.yaml` will need to be exactly the same as your clusterID field in PSO 6 `values.yaml`. Note that certain restrictions were imposed on PSO 6 clusterID, if your PSO 5 namespace does not meet PSO 6 clusterID restrictions, this upgrade tool will not work for you, please reach out for alternatives. 
+3. If your cluster has more than 500 PVCs/Attachments, it is possible the time taking to construct database content exceed the restart threshold of the controller pods, reach out to PSO team if you run into this situation.
+
+## Instruction:
+1. prepare your `values.yaml` first for PSO 6 using the PSO 5 `values.yaml`. They are formatted differently, so you won't be able to simply copy PSO 5 `values.yaml` over. Schema validation will also help to make sure you comply with PSO 6 formats. 
+2. For upgrade field in PSO 6, they are by default set to false, set it to True to enable upgrade. The upgrade takes place during the installation. 
```

Comment:
For `upgrade` field in PSO 6 `values.yaml`,  it is set to `false` by default. Set it to `true` to enable the upgrade.